spark/src/main/scala/geotrellis/spark/reproject/TileRDDReproject.scala

```diff
@@ -81,6 +82,76 @@ object TileRDDReproject {
           }
       }
 
+    val (sampleKey, BufferedTile(sampleTile, sampleGridBounds)) = bufferedTiles.first() // XXX
+    val tileCols = sampleTile.cols // XXX metadata.layout.tileCols
+    val tileRows = sampleTile.rows // XXX metadata.layout.tileRows
+    val KeyBounds(SpatialKey(keyColMin, keyRowMin), SpatialKey(keyColMax, keyRowMax)) = metadata.bounds
+    val lb = scala.collection.mutable.ListBuffer.empty[K]
+
+    if (sc != None) {
+      var col = keyColMin; while (col <= keyColMax) {
+        var row = keyRowMin; while (row <= keyRowMax) {
+          lb += sampleKey.setComponent(SpatialKey(col, row))
+          row += 1
+        }
+        col += 1
+      }
+    }
+
+    val layerInfo: (Extent, CellSize, KeyBounds[K]) =
+      (sc match {
+        case Some(sc) => sc.parallelize(lb)
+        case None => bufferedTiles.map({ case (key, _) => key })
+      })
+        .mapPartitions({ partition =>
+          val transform = Transform(crs, destCrs)
+          val inverseTransform = Transform(destCrs, crs)
+
+          partition.map { key =>
+            val innerExtent = mapTransform(key)
+            val innerRasterExtent = RasterExtent(innerExtent, sampleGridBounds.width, sampleGridBounds.height)
+            val outerGridBounds =
+              GridBounds(
+                -sampleGridBounds.colMin,
+                -sampleGridBounds.rowMin,
+                tileCols - sampleGridBounds.colMin - 1,
+                tileRows - sampleGridBounds.rowMin - 1
+              )
+            val outerExtent = innerRasterExtent.extentFor(outerGridBounds, clamp = false)
+
+            val window =
+              if(options.matchLayerExtent) {
+                sampleGridBounds
+              } else {
+                // Reproject extra cells that are half the buffer size, as to avoid
+                // any missed cells between tiles.
+                GridBounds(
+                  sampleGridBounds.colMin / 2,
+                  sampleGridBounds.rowMin / 2,
+                  (tileCols + sampleGridBounds.colMax - 1) / 2,
+                  (tileRows + sampleGridBounds.rowMax - 1) / 2
+                )
+              }
+
+            val rasterExtent = RasterExtent(outerExtent, tileCols, tileRows)
+            val windowExtent = rasterExtent.extentFor(window)
+            val windowRasterExtent = RasterExtent(windowExtent, window.width, window.height)
+            val targetRasterExtent = ReprojectRasterExtent(windowRasterExtent, transform)
+
+            (key, targetRasterExtent.extent, targetRasterExtent.cols, targetRasterExtent.rows)
+          }
+        })
+        .map({ case (key, extent, cols, rows) =>
+          (extent, CellSize(extent, cols, rows), KeyBounds(key, key))
+        })
```

Comment:
Not sure how much this will save but this can easily just be what the previous step returns and avoid the map. CPU cycle saves is CPU cycle earned, I suppose.